neurolang/probabilistic/dalvi_suciu_lift.py

```diff
@@ -163,7 +170,11 @@ def dalvi_suciu_lift(rule, symbol_table):
             for atom in extract_logic_atoms(rule)
         )
     ):
-        return TranslateToNamedRA().walk(rule)
+        free_vars = extract_logic_free_variables(rule)
+        rule = MakeExistentialsImplicit().walk(rule)
+        result = TranslateToNamedRA().walk(rule)
+        proj_cols = tuple(Constant(ColumnStr(v.name)) for v in free_vars)
+        return Projection(result, proj_cols)
```

Comment:
Isn't this covered by line 164? `convert_rule_to_ucq`

Reply:
`convert_rule_to_ucq` keeps existential quantifiers, resulting in the variable `rule` being passed to `TranslateToNamedRA` with existential quantifiers but `TranslateToNamedRA` does not translate existential predicates. I've added a test that fails here: https://github.com/tgy/NeuroLang/blob/lqp-testing/neurolang/probabilistic/tests/test_dalvi_suciu_lift.py#L489 

What do you think is the right solution?